tensorflow/lite/delegates/gpu/cl/inference_context.cc

```diff
@@ -220,7 +220,7 @@ absl::Status GetBufferAsignment(
   *use_offset_assignment = false;
   if (*is_sub_buffers_supported) {
     RETURN_IF_ERROR(AssignOffsetsToTensors(
-        *buffer_usage_records, MemoryStrategy::GREEDY_BY_SIZE,
+        *buffer_usage_records, MemoryStrategy::EQUALITY,
```

Comment:
Why does NVidia have an error with GREEDY_BY_SIZE? Won't it make any regression on different platforms?

Reply:
I'm not sure what the root cause of this is, but this bug has been present ever since OpenCL sub-buffer supported was added in TF v2.7. We use a fork of TFLite v2.7 OpenCL delegate (with MemoryStrategy::EQUALITY instead of MemoryStrategy::GREEDY_BY_SIZE) on Android as well, and we have not run into any problem. Please see this issue https://github.com/tensorflow/tensorflow/issues/53800 for more details on the bug.